.github/scripts/trymerge.py

```diff
@@ -52,6 +57,18 @@ def __init__(self, name: str, url: str, status: Optional[str]):
         self.status: Optional[str] = status
         self.jobs: JobNameToStateDict = {}
 
+class FlakyRule:
+    def __init__(self, name: str, captures: List[str]):
+        self.name: str = name
+        self.captures = captures
+
+    def matches(self, job: Optional[Dict[str, Any]]) -> bool:
+        return (
+            job is not None
+            and self.name in job.get('name', '')
+            and job.get('failure_captures', None) is not None
```

Comment:
Nit: do you need this check?  I think the one on the next line `job.get("failure_captures", [])` subsumes it